src/misc/socket.cc

```diff
@@ -727,6 +730,17 @@ ncclResult_t ncclSocketInit(struct ncclSocket* sock, union ncclSocketAddress* ad
       ret = ncclSystemError;
       goto fail;
     }
+
+    // [RCCL] Runtime socket options
+    if (rcclParamSocketReuseAddr()) {
+      int opt = 1;
+      SYSCHECKGOTO(setsockopt(sock->fd, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt)), ret, fail);
+    }
+    auto lingerParam = rcclParamSocketLinger();
```

Comment:
Since you are casting this value anyway later, I recommend using an explicit type before the subsequent `if` condition. Who knows in the future RCCL_PARAM may be changed to return double.